### PR TITLE
fix: rustup

### PR DESCRIPTION
**Pull request by fhartwig**
* enable slice_patterns feature
* add `Reflect` trait bounds where necessary

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 6a803990afe1c58f22d0e3ed326d051ad07e8bbc
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by fhartwig** [comment]
Amended commit message to make gitcop happy.

**Comment by seanmonstar** [comment]
Thanks! I wonder, if slice_patterns aren't going to be stable for beta/1.0,
we should probably not use them. I think they're only used in the
authorization header, and shouldn't be too hard to switch it to a few if
clauses.

**Comment by fhartwig** [comment]
https://github.com/rust-lang/rust/issues/23121 says that they hope to have it ready for the beta release, but I'm not sure if that still applies. I can re-write the function to avoid slice patterns if you like.

**Comment by seanmonstar** [comment]
With beta this week, those comments don't assure me. If you could remove
the patterns, awesome. Or I can a little later.

**Comment by fhartwig** [comment]
Ok, slice patterns are gone. Tests pass for me, travis seems to have trouble with github today.

File: src/header/common/authorization.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 use std::fmt;
 use std::str::{FromStr, from_utf8};
 use std::ops::{Deref, DerefMut};
+use std::marker::Reflect;
```
Comment (reem) [inline src/header/common/authorization.rs]: Reflect shouldn't be needed in this file as far as I can tell. What error comes up if you remove these bounds?

Comment (reem) [inline src/header/common/authorization.rs]: I see, since our Header traits should require Reflect.

**Comment by reem** [comment]
The Header downcasting stuff all needs `Reflect` bounds similar to `Any` and `NetworkStream` to be safe.

**Comment by Ryman** [comment]
According to [this](http://doc.rust-lang.org/std/marker/trait.Reflect.html) the bound should be `Any` rather than `Reflect + 'static` as `Any` implies that?

File: src/header/common/authorization.rs (hunk 1 above)
Comment (fhartwig) [inline src/header/common/authorization.rs]:
```
error: the trait `core::marker::Reflect` is not implemented for the type `S` [E0277]
```
(in lines 26 and 47)

**Comment by reem** [comment]
@Ryman you're right. The existing `Any` bound should be sufficient.

**Pull request merged**